src/main/scala/pureconfig/ConfigConvert.scala

```diff
@@ -6,227 +6,199 @@
  */
 package pureconfig
 
-import pureconfig.conf._
-import pureconfig.conf.namespace._
+import com.typesafe.config._
 import shapeless._
 import shapeless.labelled._
 
+import scala.collection.JavaConverters._
 import scala.collection.generic.CanBuildFrom
 import scala.language.higherKinds
 import scala.util.{ Failure, Success, Try }
 
+import java.net.URL
+import pureconfig.ConfigConvert.{ fromString, stringConvert }
+
 /**
- * Trait for conversion between `T` and `RawConfig` where `T` is a "complex" type. For "simple"
- * types have a look at [[StringConvert]]
+ * Trait for conversion between `T` and `ConfigValue`.
  */
 trait ConfigConvert[T] {
   /**
-   * Convert the given configuration into an instance of `T` if possible
+   * Convert the given configuration into an instance of `T` if possible.
    *
    * @param config The configuration from which load the config
-   * @param namespace The base namespace to use for the conversion. This should be used as base
-   *                  for the name of the fields of `T`. For instance, given a `case class Foo(i: Int)`
-   *                  and a configuration `conf = Map("foo.i", "1")`, then `from(conf, "foo")` will return
-   *                  `Success(Foo(1))` while `from(conf, "")` will return `Failure`
    * @return `Success` of `T` if the conversion is possible, `Failure` with the problem if the
    *         conversion is not
    */
-  def from(config: RawConfig, namespace: String): Try[T]
+  def from(config: ConfigValue): Try[T]
 
   /**
-   * Converts a type `T` to a `RawConfig` using the `namespace` as base namespace.
+   * Converts a type `T` to a `ConfigValue`.
    *
    * @param t The instance of `T` to convert
-   * @param namespace The base namespace. For instance, given case `class Foo(i: Int)`, then
-   *                  `to(Foo(2), "base.namespace")` will return `Map("base.namespace.i", "2")`
-   * @return The `RawConfig` obtained from the `T` instance
+   * @return The `ConfigValue` obtained from the `T` instance
    */
-  def to(t: T, namespace: String): RawConfig
+  def to(t: T): ConfigValue
 }
 
 object ConfigConvert extends LowPriorityConfigConvertImplicits {
   def apply[T](implicit conv: ConfigConvert[T]): ConfigConvert[T] = conv
 
+  private def fromFConvert[T](fromF: String => T): ConfigValue => Try[T] =
+    config => {
+      Try(fromF(config.valueType() match {
+        case ConfigValueType.STRING => config.unwrapped().toString
+        case _ => config.render(ConfigRenderOptions.concise())
+      }))
+    }
+
+  def fromString[T](fromF: String => T): ConfigConvert[T] = new ConfigConvert[T] {
```

Comment:
I think  we can postpone this discussion for now, this PR is big enough :-). Let's just keep this idea in consideration for future releases, I think it could be useful.